File: controller/src/main/java/yeetcd/controller/pipeline/Parameters.java
package yeetcd.controller.pipeline;

import java.util.Collections;
import java.util.Map;

public record Parameters(Map<String, Parameter> parameters) {

    public static Parameters fromMap(Map<String, Parameter> map) {
        return new Parameters(map);
    }

    public static Parameters empty() {
        return new Parameters(Collections.emptyMap());
    }
}

File: internal/core/pipeline/parameters.go
package pipeline

// Parameters wraps a map of parameter definitions
type Parameters map[string]Parameter

// EmptyParameters creates empty Parameters
func EmptyParameters() Parameters {
	return make(Parameters)
}

// ParametersFromMap creates Parameters from a map
func ParametersFromMap(m map[string]Parameter) Parameters {
	return Parameters(m)
}
